### PR TITLE
Add sidebar_widgets and notebook_preamble params

### PR DESCRIPTION
**Pull request by ahuang11**
To allow deploying a notebook session in a custom way, but maybe this can be generalized to a custom action callback.

Closes https://github.com/holoviz/lumen/issues/591

### CODE REVIEW

File: lumen/ai/views.py [hunk 1]
```diff
@@ -122,10 +122,16 @@ async def _render_component(self):
...
-        memory["current_table"] = self.name
-        source.add_table(self.name, self.spec)
+        if hasattr(source, "add_table"):
+            # self.name -> SQLOutput02697?
+            memory["current_table"] = self.name
+            source.add_table(self.name, self.spec)
+            table = self.name
+        else:
+            table = memory["current_table"]
+
```
Comment (ahuang11) [inline lumen/ai/views.py:132]:
```suggestion
        source = memory["current_source"]
        memory["current_table"] = self.name
        source.add_table(self.name, self.spec)
```

File: lumen/ai/views.py [hunk 2]
```diff
@@ -122,10 +122,16 @@ async def _render_component(self):
...
         try:
-            pipeline = Pipeline(source=source, table=self.name)
+            pipeline = Pipeline(source=source, table=table)
```
Comment (ahuang11) [inline lumen/ai/views.py:134]:
```suggestion
            pipeline = Pipeline(source=source, table=self.name)
```

File: lumen/ai/views.py [hunk 3]
```diff
@@ -121,6 +121,7 @@ async def _render_component(self):
                 value=True, name="Executing SQL query...", height=50, width=50
             )
 
+        source = memory["current_source"]
```
Comment (ahuang11) [inline lumen/ai/views.py:124]:
```suggestion
```

**Comment by ahuang11** [comment]
Okay I think I'm going with `sidebar_widgets` because it gives the user maximum flexibility.

Initially, I was thinking of copying functionality from [ChatInterface.button_properties](https://github.com/holoviz/panel/blob/86eff50dbee38c8665fca5eb66ddf3b3e889bc84/panel/chat/interface.py#L117), but over time, I found that to be a bit limiting (e.g. what if the user wants to use a ToggleButton or FileDownload widget instead?).

**Review by philippjfr — APPROVED** [review]

**Pull request merged**